src/Engine/Renderer/Material/MaterialConverters.hpp

```diff
@@ -0,0 +1,83 @@
+#ifndef RADIUMENGINE_MATERIALCONVERTERS_HPP
+#define RADIUMENGINE_MATERIALCONVERTERS_HPP
+
+#include <Engine/RaEngine.hpp>
+
+#include <Core/File/MaterialData.hpp>
+
+#include <functional>
+#include <string>
+
+namespace Ra {
+  namespace Engine {
+    class Material;
+  }
+}
+
+
+/// A material converter is a couple <std::string, std::function<Ra::Engine::Material*(Ra::Asset::MaterialData*)>
+/// The string gives the mname of the material, the function is watever is compatible wit std::function :
+///     - a lambda
+///     - a functor
+///     - a function with bind parameteres ....
+/// The function is in charge of converting a concrete Ra::Asset::MaterialData* to a concrete Ra::Engine::Material*
+/// according to the type of material described by the string ...
+/**
+ * Instruction on how to extend the material system
+ */
+///////////////////////////////////////////////
+////        Material converter system       ///
+///////////////////////////////////////////////
+
+namespace Ra {
+  namespace Engine {
+    namespace MaterialConverterSystem {
+
+      using AssetMaterialPtr = const Ra::Asset::MaterialData *;
+      using RadiumMaterialPtr = Ra::Engine::Material *;
+      using ConverterFunction = std::function<RadiumMaterialPtr(AssetMaterialPtr)>;
+
+      /** register a new material converter
+      *  @return true if converter added, false else (e.g, a converter with the same name exists)
+      */
+      bool registerMaterialConverter(const std::string &name, ConverterFunction converter);
+
```

Comment:
Why not using the Manager pattern or put these methods directly in the RadiumEngine?

Reply:
Manager pattern rely on the implementation of a singleton pattern that is not really robust in this case. More, using the manager pattern will require to have access to then "Engine" Object and thus, might restrict the usage of material converters for Radium rendering applications.